### ISSUE TITLE
Allow ThemeIcon in WebviewPanel.iconPath

### ISSUE THREAD

**Issue by ridiculous-thought** [microsoft/vscode#90616]
Issue Type: <b>Feature Request</b>

You can only pass an Uri or an object containing Uris to WebviewPanel.iconPath, while iconPath on QuickInputButton and TreeItem also supports ThemeIcon. This would increase codicon reusability.

VS Code version: Code 1.42.0 (ae08d5460b5a45169385ff3fd44208f431992451, 2020-02-06T10:51:34.058Z)
OS version: Windows_NT x64 10.0.18362

**Comment by mjbvz** [comment]
@bpasero I believe we should change how editors get their icons to support this. My proposal is to add a `IEditorInput.icon` property that lets editor inputs set their icon directly instead of going through css rules based on the resource. This could also be used for our welcome page and a few other places in VS Code core that have to use css for this today: https://github.com/microsoft/vscode/blob/ceb8d830f88e516ff7074127189ef01497b2feb2/src/vs/workbench/contrib/welcome/page/browser/welcomePage.css#L223

**Comment by bpasero** [comment]
Works for me.

**Comment by mjbvz** [comment]
@aeschli Adding you for help with #92122. Let's try looking into this in May

**Comment by yiqing-zhao** [comment]
Any update? Looking forward to use ThemeIcons in webviews!

**Comment by nora-soderlund** [comment]
What is the current workaround?

**Comment by bpasero** [comment]
Now @benibenj introduced support for editors to return a `ThemeIcon`:

https://github.com/microsoft/vscode/blob/b4870983b477caabcb6ef75b5bed4aa11592bfaf/src/vs/workbench/common/editor/editorInput.ts#L167-L173

So I think this can move along, unless I miss something.

**Comment by nora-soderlund** [comment]
What about setting a ThemeIcon to an editor? @bpasero 

I might be missing something.

**Comment by bpasero** [comment]
Yeah I muss the bigger picture of this issue, I was just to mention we made some progress in core for the reading side.

**Comment by mjbvz** [comment]
To verify, try setting a theme icon as the webviewPanels.iconPath property. You can use this webview sample: https://github.com/microsoft/vscode-extension-samples/tree/main/webview-sample
